src/main/groovy/org/gradle/android/workarounds/RoomSchemaLocationWorkaround.groovy

```diff
@@ -127,40 +155,115 @@ class RoomSchemaLocationWorkaround implements Workaround {
         return compilerPluginOptions.subpluginOptionsByPluginId[KAPT_SUBPLUGIN_ID]
     }
 
-    private void configureSchemaLocationOutputs(Task task) {
-        def processorOptions = getCompilerPluginOptions(task)
-        processorOptions.each { option ->
-            if (option.key == ROOM_SCHEMA_LOCATION) {
-                def relativePath = option.value
-                def schemaDestinationDir = task.project.file(relativePath)
-                def taskSpecificSchemaDir = getTaskSpecificSchemaDir(task)
+    static def getEncodedCompilerPluginOptions(Task task) {
+        def pluginOptionsField = task.class.superclass.getDeclaredField("pluginOptions")
+        pluginOptionsField.setAccessible(true)
```

Comment:
is this required? i always that that groovy could access private fields without reflection.

Reply:
It's possible that groovy -> groovy might allow accessing private fields, but groovy -> java definitely won't work without explicitly making it accessible.